File: Java/bir işlem/entity/GameData.java
package entity;

import java.util.ArrayList;
import java.util.List;

/**
 * Oyun verilerinin tutulduğu varlık sınıfı.
 * 
 * @apiNote Oyun verilerini tutar
 * @category Varlık Sınıfı
 */
public class GameData {
    private List<Integer> numberList;
    private int targetNumber;
    private List<String> npcNames;

    /**
     * GameData varlık sınıfını parametresiz oluşturur
     */
    public GameData() {
        numberList = new ArrayList<>();
        npcNames = new ArrayList<>();
        npcNames.add("Darth Maul");
        npcNames.add("Darth Vader");
        npcNames.add("Darth Sidious");
        npcNames.add("Darth Tyranus");
        npcNames.add("Darth Bane");
        npcNames.add("Darth Plagus");
    }

    /**
     * Hesaplarda kullanılacak sayıların listesini verir
     * 
     * @return numberList -> Sayı listesi (ArrayList[int])
     */
    public List<Integer> getNumberList() {
        return numberList;
    }

    /**
     * Hesaplarda kullanılmak üzere sayı listesine bir tam sayı ekler
     * 
     * @param number -> Tam sayı (int)
     */
    public void setNumberList(int number) {
        this.numberList.add(number);
    }

    /**
     * Hesaplamalar sonucu ulaşılmak istenen sayıyı verir
     * 
     * @return targetNumber -> Hedeflenen sayı (int)
     */
    public int getTargetNumber() {
        return targetNumber;
    }

    /**
     * Hesaplamalar sonucu ulaşılmak istenen sayıyı ayarlar
     * 
     * @param targetNumber -> Hedeflenen sayı (int)
     */
    public void setTargetNumber(int targetNumber) {
        this.targetNumber = targetNumber;
    }

    /**
     * MPC (Non-Player Character) isim listesini verir
     * 
     * @return npcNames -> NPC isimleri (ArrayList[String])
     */
    public List<String> getNpcNames() {
        return npcNames;
    }

}

File: Python/bir işlem/entity/game_data.py
from tables import Unknown


class GameData:
    """Oyun verilerinin tutulduğu varlık sınıfı
    @category: Entity Class"""

    def __init__(self) -> None:
        """Oyun verilerinin tutulduğu varlık sınıfı
    @category: Entity Class"""
        self._number_list = []
        self._target_number = 0
        self._npc_names = [
            "Darth Maul",
            "Darth Vader",
            "Darth Sidious",
            "Darth Tyranus",
            "Darth Bane",
            "Darth Plaguis"
        ]

    @property
    def number_list(self) -> list[int]:
        """Hesaplamada kullanılacak sayıların listesini geri döndürür

        Returns:
            number_list (list[int]): hesaplamalarda kullanılacak sayıların listesi
        """
        return self._number_list

    @number_list.setter
    def number_list(self, nlist: list[int]) -> None:
        """Hesaplamada kullanılacak sayıların listesini ayarlar

        Args:
            list (list[int]): Sayı listesi
        """
        self._number_list = nlist

    @property
    def target_number(self) -> int:
        """Hesaplamalar sonucu ulaşılmak istenen sayıyı geri döndürür

        Returns:
            target_number (int): ulaşılmak istenen sayı"""
        return self._target_number

    @target_number.setter
    def target_number(self, number: int) -> None:
        """Hesaplamalar sonucu ulaşılmak istenen sayıyı ayarlar

        Args:
            target_number (int): ulaşılmak istenen sayı
        """
        self._target_number = number

    @property
    def npc_names(self) -> list[str]:
        """NPC (Non-Player Character) isim listesini geri döndürür

        Returns:
            npc_names (list[str]): NPC isimleri
        """
        return self._npc_names
